### PR TITLE
JBPM-7668 (Stunner) - Align and distribute doesn't work properly

### PR DESCRIPTION
**Pull request by Josephblt**
https://issues.jboss.org/browse/JBPM-7668

There was an issue with a flag that signals if the control's attribute
changes should be handled. It was causing problems when using undo/redo
and probably other actions as well.

### CODE REVIEW

**Comment by Josephblt** [comment]
Hey @romartin, @hasys. If you guys could please review this I'd appreciate. Thanks.

**Comment by tiagodolphine** [comment]
Jenkins execute full downstream build

**Comment by tiagodolphine** [comment]
Hi, @Josephblt I tested the generated war, but it seems there still persists the issues regarding aligning and distribute.

**Comment by Josephblt** [comment]
It's important to notice. There is another issue:
https://issues.jboss.org/browse/JBPM-7715 
This is the PR that fixed this issue:
https://github.com/kiegroup/lienzo-core/pull/60

This PR does not fix those first time A&D problems. 
Because of this, it is hard to keep track on what is the nature is undo/redo related, or a first time error. 
To avoid this I recommend testing with both fixes. All errors are easily reproducible on Stunner Standalone Showcase.

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by Josephblt** [comment]
Hey @hasys. Sure, I'll add more tests.

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by Josephblt** [comment]
Hey guys @tiagodolphine, @hasys, @romartin. All done. Tests and some extra fixes. If you could review-merge I'be glad. Thanks.

File: src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/AlignAndDistributeControlImpl.java [hunk 1]
```diff
@@ -526,15 +521,15 @@ private void indexOn(AlignAndDistributeControl handler)
         if (handler != null && handler.isIndexed())
         {
             m_alignAndDistribute.indexOnWithoutChangingStatus(handler);
-            ((AlignAndDistributeControlImpl)handler).m_indexedButRemoved = true;
+            ((AlignAndDistributeControlImpl)handler).m_indexedButRemoved = false;
```
Comment (romartin) [inline src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/AlignAndDistributeControlImpl.java]: hey @Josephblt , tbh a bit lost about this change, I don't know this code quite well... but why this change?

**Comment by Josephblt** [comment]
This flags controls if this component is going to be updated whenever refresh is called. It should be set to false when you end drag to allow it to be updated again.

**Comment by hasys** [comment]
Hi @jhrcek,

there are a lot of failing tests in KIE Server integration:

<details>
  <summary>Click to see full list</summary>
  <pre>org.kie.server.integrationtests.controller.WebSocketKieControllerStartupIntegrationTest.testRegisterKieServerAfterStartup[0: JAXB KieServicesConfiguration{transport=REST, serverUrl='http://localhost:43676/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.controller.WebSocketKieControllerStartupIntegrationTest.testContainerCreatedAfterStartup[0: JAXB KieServicesConfiguration{transport=REST, serverUrl='http://localhost:43676/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.controller.WebSocketKieControllerStartupIntegrationTest.testTurnOffKieServerAfterShutdown[0: JAXB KieServicesConfiguration{transport=REST, serverUrl='http://localhost:43676/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.controller.WebSocketKieControllerStartupIntegrationTest.testContainerDisposedAfterStartup[0: JAXB KieServicesConfiguration{transport=REST, serverUrl='http://localhost:43676/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.controller.WebSocketKieControllerStartupIntegrationTest.testRegisterKieServerAfterStartup[1: JSON KieServicesConfiguration{transport=REST, serverUrl='http://localhost:43676/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.controller.WebSocketKieControllerStartupIntegrationTest.testContainerCreatedAfterStartup[1: JSON KieServicesConfiguration{transport=REST, serverUrl='http://localhost:43676/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.controller.WebSocketKieControllerStartupIntegrationTest.testTurnOffKieServerAfterShutdown[1: JSON KieServicesConfiguration{transport=REST, serverUrl='http://localhost:43676/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.controller.WebSocketKieControllerStartupIntegrationTest.testContainerDisposedAfterStartup[1: JSON KieServicesConfiguration{transport=REST, serverUrl='http://localhost:43676/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.ProcessServiceIntegrationTest.testSignalContainer[1: JSON KieServicesConfiguration{transport=REST, serverUrl='http://localhost:39129/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.ProcessServiceIntegrationTest.testSignalContainer[2: XSTREAM KieServicesConfiguration{transport=REST, serverUrl='http://localhost:39129/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.ProcessServiceIntegrationTest.testSignalContainer[3: JAXB KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.jbpm.ProcessServiceIntegrationTest.testSignalContainer[4: JSON KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.jbpm.ProcessServiceIntegrationTest.testSignalContainer[5: XSTREAM KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.jbpm.SLAComplianceIntegrationTest.testSLAonUserTaskViolated[0: JAXB KieServicesConfiguration{transport=REST, serverUrl='http://localhost:39129/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.SLAComplianceIntegrationTest.testSLAonProcessViolated[1: JSON KieServicesConfiguration{transport=REST, serverUrl='http://localhost:39129/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.SLAComplianceIntegrationTest.testSLAonUserTaskViolated[1: JSON KieServicesConfiguration{transport=REST, serverUrl='http://localhost:39129/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.SLAComplianceIntegrationTest.testSLAonUserTaskViolated[2: XSTREAM KieServicesConfiguration{transport=REST, serverUrl='http://localhost:39129/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.SLAComplianceIntegrationTest.testSLAonUserTaskViolated[3: JAXB KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.jbpm.SLAComplianceIntegrationTest.testSLAonProcessViolated[4: JSON KieServicesConfiguration{transport=JMS, serverUrl='null'}]</pre>
</details>

Jenkins execute full downstream build

**Comment by romartin** [comment]
> This flags controls if this component is going to be updated whenever refresh is called. It should be set to false when you end drag to allow it to be updated again.

Ok @Josephblt , so if we're going to finally NOT re-index on each click this change about the flag is not necessary, doesn't apply anymore, isn't it?

**Comment by Josephblt** [comment]
Hey @romartin. Actually the line is needed. I'll try to explain better.

[JBPM-7668](https://github.com/kiegroup/lienzo-core/pull/61) (this one) - This PR fixes problems when using undo/redo was used and some other specific scenarios.
[JBPM-7715](https://github.com/kiegroup/lienzo-core/pull/60) - This fixes problems when you drag a control for the first time.

Both [JBPM-7668](https://github.com/kiegroup/lienzo-core/pull/61) and [JBPM-7715](https://github.com/kiegroup/lienzo-core/pull/60). rely on the indexation as it happens now, (when some attribute change event is fired, for example).

[JBPM-6920](https://github.com/kiegroup/lienzo-core/pull/62) - This is the PR that changes indexation behavior. All controls are indexed when drag starts. If this PR is merged the fixes mentioned above are not needed.

From what I understood we should try and merge:
[JBPM-7668](https://github.com/kiegroup/lienzo-core/pull/61) (this one)
[JBPM-7715](https://github.com/kiegroup/lienzo-core/pull/60)
If things don't stabilize after this we should try and merge the always re-index PR ([JBPM-6920](https://github.com/kiegroup/lienzo-core/pull/62))

**Comment by hasys** [comment]
Hi @romartin, @Josephblt,

yes I tested two PRs together (see list of PRs and tests for it below) and didn't found further issues with Align and Distributions. Also changes introduced by two PRs looks safe for me. But yes, third one probably should be postponed for now.

- https://github.com/kiegroup/lienzo-core/pull/61
- https://github.com/kiegroup/lienzo-tests/pull/45
- https://github.com/kiegroup/lienzo-core/pull/60
- https://github.com/kiegroup/lienzo-tests/pull/44
